gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/shared_policy_group/use_case/DeleteSharedPolicyGroupUseCase.java

```diff
@@ -36,8 +36,13 @@ public class DeleteSharedPolicyGroupUseCase {
     private final SharedPolicyGroupCrudService sharedPolicyGroupCrudService;
     private final SharedPolicyGroupHistoryCrudService sharedPolicyGroupHistoryCrudService;
     private final AuditDomainService auditService;
+    private final UndeploySharedPolicyGroupUseCase undeploySharedPolicyGroupUseCase;
 
     public Output execute(Input input) {
+        undeploySharedPolicyGroupUseCase.execute(
```

Comment:
UseCases must not call on other use cases. 
So here what about juste re impl publishEvent to undeploy SPG link in undeployUseCase
But without creation of new audit ext ... which are not the usual case here